stopes/ui/seamlisten/backend/app/fileviewer.py

```diff
@@ -105,52 +103,56 @@ def open_zip_file(file: str, start_idx: int, end_idx: int) -> tp.List[LineResult
 
 
 @router.post("/annotations/")
-def get_annotations(
-    query: AnnotationQuery,
-) -> tp.List[LineResult]:
+def get_annotations(query: AnnotationQuery) -> tp.List[LineResult]:
     path = query.gz_path.strip()
-    print(path, query.start_idx, query.end_idx)
-    if path.endswith(".tsv.gz") or path.endswith(".tsv"):
+    resolved_path = Path(path).expanduser().resolve()
+
+    if resolved_path.suffixes in ([".gz"], [".tsv"]):
         try:
-            return open_segment_tsv(path, query.start_idx, query.end_idx)
+            return open_segment_tsv(str(resolved_path), query.start_idx, query.end_idx)
         except FileNotFoundError:
             raise HTTPException(
                 status_code=404,
-                detail="""
-                File not found
-            """,
+                detail="File not found",
             )
-    elif path.endswith(".zip"):
-        if not Path(path).exists():
+    elif resolved_path.suffixes == [".zip"]:
+        if not resolved_path.exists():
             raise HTTPException(status_code=404, detail="File not found")
-        return open_zip_file(path, query.start_idx, query.end_idx)
+        return open_zip_file(str(resolved_path), query.start_idx, query.end_idx)
     else:
         raise ValueError("Unknown format")
 
 
 @router.post("/general/")
 async def general_query(query: DefaultQuery) -> Response:
-    query_path = query.gz_path.strip()
-    print(query_path.split(" "))
-    if query_path.endswith(".tsv.gz") or query_path.endswith(".zip"):
+    query_path_str = query.gz_path.strip()
+    query_path = Path(query_path_str).expanduser().resolve()
```

Comment:
will this work when the path is one of the other formats (which are not paths?), like:
```
a/b/c 100 10000
a/b/c|0|231231
```

Reply:
yes it will work normally

Reply:
@Mortimerp9 
can you re-review this PR